cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -212,7 +212,7 @@ public void onBecomeLeaderFromFollower(Message message, NotificationContext cont
         // sanity check no existing Leader for up to 59 seconds
         for (int i = 0; i < 60; ++i) {
           view = admin.getResourceExternalView(cluster, resourceName);
-          stateMap = view.getStateMap(partitionName);
+          stateMap = Utils.getStateMap(view, partitionName);
```

Comment:
For this `onBecomeLeaderFromFollower`, when `stateMap` is empty, it will `break` the for loop and treat it as no Leader. Then, it will skip a few section of code later on and eventually promote itself as Leader, is this expected?

(can you add similar explanation what would happen when `{}` returned compared to null, in other places of the code change)

Reply:
Good question. For this case, this logic is to wait for up to 1min to ensure there is **no existing leader** in the shard, so we won't have two leaders for a shard at the same time. 

If somehow the shard is empty (no leader or followers), it could indicate the shard is temporarily down, and helix decide to promote a leader immediately, but then the shard doesn't have live instances at that point. This is a edge case, but when it happens, we ave a few choices:
- just crash and become a ERROR shard, which is the current behavior without the null check
- continue to become a leader

I think the later seems more like the intended behavior: a state transition is triggered for a shard to go from follower to leader, and the current shard is empty, so we should let it go through. 
Thoughts?

Reply:
Also added similar explanation in other parts

Reply:
If the actual zk data is `shard is empty (no leader or followers)`, then, I agree with you, we should let helix promote this current replica to Leader immediately to at least recover the availability.